### PR TITLE
Delete vector of pointers before exit

### PR DESCRIPTION
**Pull request by peterfpeterson**
Valgrind found a memory issue [here](http://builds.mantidproject.org/view/Valgrind/job/valgrind_core_packages/1137/valgrindResult/pid=28936/) in `TimeSeriesPropertyTest`. This PR just deletes the vector of pointers before exiting the test in an attempt to make Valgrind happy.

**To test:**

The builds should all still pass.

*There is no associated issue.*

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Kernel/test/TimeSeriesPropertyTest.h [hunk 1]
```diff
@@ -962,6 +962,11 @@ class TimeSeriesPropertyTest : public CxxTest::TestSuite {
     for (size_t i = 0; i < 10; ++i) {
       TS_ASSERT_EQUALS(outputs[i]->size(), 2);
     }
+
+    // cleanup
+    for (auto it = outputs.begin(); it != outputs.end(); ++it)
```
Comment (quantumsteve) [inline Framework/Kernel/test/TimeSeriesPropertyTest.h]: Change to a range-based for loop.

File: Framework/Kernel/test/TimeSeriesPropertyTest.h [hunk 2]
```diff
@@ -962,6 +962,11 @@ class TimeSeriesPropertyTest : public CxxTest::TestSuite {
...
+      delete (*it);
+    outputs.clear();
```
Comment (quantumsteve) [inline Framework/Kernel/test/TimeSeriesPropertyTest.h]: Is this already done in the destructors?

**Review by quantumsteve — CHANGES_REQUESTED** [review]
Reasonable fix to this memory leak. Should we add an issue to replace manual memory management with RAII?

File: Framework/Kernel/test/TimeSeriesPropertyTest.h (hunk 2 above)
Comment (peterfpeterson) [inline Framework/Kernel/test/TimeSeriesPropertyTest.h]: Not according to valgrind

**Review by quantumsteve — APPROVED** [review]

**Comment by quantumsteve** [comment]
LGTM :shipit:

**Pull request merged**